easybuild/tools/filetools.py

```diff
@@ -2680,8 +2684,10 @@ def get_source_tarball_from_git(filename, target_dir, git_config):
     if not url:
         raise EasyBuildError("url not specified in git_config parameter")
 
-    if not filename.endswith('.tar.gz'):
-        raise EasyBuildError("git_config currently only supports filename ending in .tar.gz")
+    file_ext = find_extension(filename, required=False)
+    if file_ext:
+        print_warning(f"Ignoring extension of filename '{filename}' set in git_config parameter")
```

Comment:
Isn't this going to require cleaning up a whole bunch of easyconfigs?

Reply:
Rather than printing a warning, we should enforce the use of `.tar.xz` as extension here (to "ease" the transition, in some way)

Reply:
as discussed, the archive extension is no longer hardcoded or ignored. This PR now improves `get_source_tarball_from_git` to support more extensions than what we do in 4.9. See https://github.com/easybuilders/easybuild-framework/pull/4660/commits/1c44ad74338721435b272f1ccde1b58c26965045